tests/components/starlink/test_config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""Test the Starlink config flow."""
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.starlink.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+
+from .patchers import DEVICE_FOUND_PATCHER, NO_DEVICE_PATCHER, SETUP_ENTRY_PATCHER
+
+from tests.common import MockConfigEntry
+
+
+async def test_flow_user_fails_no_dishy(hass: HomeAssistant) -> None:
+    """Test user initialized flow fails when Starlink is available."""
+    user_input = {CONF_IP_ADDRESS: "192.168.100.1:9200"}
+
+    with NO_DEVICE_PATCHER, SETUP_ENTRY_PATCHER:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+        )
+        await hass.async_block_till_done()
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=user_input,
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["errors"]
+    hass.config_entries.flow.async_abort(result["flow_id"])
+    await hass.async_block_till_done()
```

Comment:
We shouldn't abort the flow here.  We should validate they can still get to success state by entering the correct data after we get to the error state here.

ex the test is done when we reach a `data_entry_flow.FlowResultType.CREATE_ENTRY` or `data_entry_flow.FlowResultType.ABORT` state

Reply:
I have updated the test, what do you think?